core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -48,6 +48,19 @@ package object config {
     .bytesConf(ByteUnit.MiB)
     .createOptional
 
+  private[spark] val DRIVER_LOG_DFS_DIR =
+    ConfigBuilder("spark.driver.log.dfsDir").stringConf.createOptional
+
+  private[spark] val DRIVER_LOG_LAYOUT =
+    ConfigBuilder("spark.driver.log.layout")
+      .stringConf
+      .createWithDefault("%d{yy/MM/dd HH:mm:ss.SSS} %t %p %c{1}: %m%n")
+
+  private[spark] val DRIVER_LOG_SYNCTODFS =
```

Comment:
With the removal of the YARN code, does it even make sense to have this option? Seems like setting `DRIVER_LOG_DFS_DIR` would be enough to enable this.

Only advantage of it is that it makes it simpler to disable the logger if the default config file has it enabled.

Reply:
Yes, as you said I prefer to keep the functionality of disabling/enabling separate from log dir. But I do see that removing this reduces the number of configurations.

I am ok either ways. Please let me know if I should remove this.

Reply:
Let's leave it. But there's some usage of the enabled flag in your code that you can still clean up.